docs/subnets/introduction-to-vm.md

```diff
@@ -0,0 +1,333 @@
+# Introduction to Virtual Machines
+
+## Overview
+
+This is part of a series of tutorials for building a Virtual Machine (VM):
+
+- Introduction to Virtual Machines (this article)
+- [How to Build a Simple VM](./create-a-virtual-machine-vm.md)
+- [How to Build a Complex VM](./create-a-vm-blobvm.md)
+
+A Virtual Machine is a blueprint for a blockchain and a blockchain instantiates from a VM, very much similar to a class-object relationship. A Virtual Machine can define following: issuance of transactions, transaction types, block structure, block building algorithm, keeping pending transactions in the mempool, gossiping transactions to the connected nodes, etc.
+
+## Blocks and State
+
+Virtual Machines can be broken down into 2 components: blocks and state. The functionality provided by VMs is to:
+
+- Define the representation of a blockchain's state
+- Represent the operations in that state
+- Apply the operations in that state
+
+Each block in the blockchain contains a set of state transitions. Each block is applied in order from the blockchain's initial genesis block to its last accepted block to reach the latest state of the blockchain.
+
+## Blockchain
+
+A blockchain has 2 components: **Consensus Engine** and **Virtual Machine**. VMs mainly deal with the implementation related to the block's structure: building and parsing. The Consensus Engine helps in reaching consensus on the block built by the VM. Here is a brief overview:
```

Comment:
Do you have a circular definition? A VM can initiate multiple blockchains. Here we say `A blockchain has 2 components: **Consensus Engine** and **Virtual Machine**.`. We need a better explanation.

Reply:
Can we use this - 

A blockchain relies on 2 components: **Consensus Engine** and **Virtual Machine**. VM is the blueprint for a blockchain that mainly deals with...

Reply:
Sounds good to me.